validphys2/src/validphys/n3fit_data.py

```diff
@@ -102,23 +102,68 @@ def replica_luxseed(replica, luxseed):
     return replica_nnseed(replica, luxseed)
 
 
-class _TrMasks(TupleComp):
+class _Masks(TupleComp):
     """Class holding the training validation mask for a group of datasets
     If the same group of dataset receives the same trvlseed then the mask
     will be the same.
     This class holds said information so it can be reused easily, i.e.,
     ``group_name`` and ``seed`` define the ``masks``.
+
+    In case of the diagonal basis, the ``rotation`` is a tuple of the eigenvalues and
+    eigenvectors of the fitting covariance matrix.
     """
 
-    def __init__(self, group_name, seed, masks=None):
-        self.masks = masks
+    def __init__(
+        self, group_name, seed, tr_masks, vl_masks, diagonal_basis=False, eig_vals=None, u=None
```

Comment:
I think u is not self-explanatory enough, I think it should be added to the docstr or perhaps call it diagonal_rotation or something here in the class (then later on u = mask.diagonal rotation is self explanatory)